crates/core/src/db/relational_db.rs

```diff
@@ -76,6 +82,80 @@ pub struct RelationalDB {
     _lock: LockFile,
 }
 
+struct SnapshotWorker {
+    _handle: tokio::task::JoinHandle<()>,
+    /// Send end of the [`Self::snapshot_loop`]'s `trigger` receiver.
+    ///
+    /// Send a message along this queue to request that the `snapshot_loop` asynchronously capture a snapshot.
+    request_snapshot: mpsc::UnboundedSender<()>,
+}
+
+impl SnapshotWorker {
+    fn new(committed_state: Arc<RwLock<CommittedState>>, repo: Arc<SnapshotRepository>) -> Self {
+        let (request_snapshot, trigger) = mpsc::unbounded();
+        let handle = tokio::spawn(Self::snapshot_loop(trigger, committed_state, repo));
+        SnapshotWorker {
+            _handle: handle,
+            request_snapshot,
+        }
+    }
+
+    /// The snapshot loop takes a snapshot after each `trigger` message received.
+    async fn snapshot_loop(
+        mut trigger: mpsc::UnboundedReceiver<()>,
+        committed_state: Arc<RwLock<CommittedState>>,
+        repo: Arc<SnapshotRepository>,
+    ) {
+        while let Some(()) = trigger.next().await {
+            let committed_state = committed_state.clone();
+            let repo = repo.clone();
+            tokio::task::spawn_blocking(move || Self::take_snapshot(&committed_state, &repo))
+                .await
+                .unwrap();
+        }
+    }
+
+    fn take_snapshot(committed_state: &RwLock<CommittedState>, snapshot_repo: &SnapshotRepository) {
```

Comment:
In reference to the scheduling comment above, I would imagine that this function would be added to the `RelationalDB` type.